src/main/java/org/opensearch/knn/index/codec/KNN990Codec/KNN990QuantizationStateReader.java

```diff
@@ -0,0 +1,146 @@
+import org.opensearch.knn.quantization.models.quantizationState.MultiBitScalarQuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.OneBitScalarQuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationStateReadConfig;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Reads quantization states
+ */
+public final class KNN990QuantizationStateReader {
+
+    /**
+     * Read quantization states and return list of fieldNames and bytes
+     * File format:
+     * Header
+     * QS1 state bytes
+     * QS2 state bytes
+     * Number of quantization states
+     * QS1 field number
+     * QS1 state bytes length
+     * QS1 position of state bytes
+     * QS2 field number
+     * QS2 state bytes length
+     * QS2 position of state bytes
+     * Position of index section (where QS1 field name is located)
+     * -1 (marker)
+     * Footer
+     *
+     * @param state the read state to read from
+     */
+    public static Map<String, byte[]> read(SegmentReadState state) throws IOException {
+        String quantizationStateFileName = getQuantizationStateFileName(state);
+        Map<String, byte[]> readQuantizationStateInfos = new HashMap<>();
+
+        try (IndexInput input = state.directory.openInput(quantizationStateFileName, IOContext.READ)) {
+            CodecUtil.retrieveChecksum(input);
+
+            int numFields = getNumFields(input);
+
+            // Read each field's metadata from the index section and then read bytes
+            for (int i = 0; i < numFields; i++) {
+                int fieldNumber = input.readInt();
+                int length = input.readInt();
+                long position = input.readVLong();
+                byte[] stateBytes = readStateBytes(input, position, length);
+                String fieldName = state.fieldInfos.fieldInfo(fieldNumber).getName();
+                readQuantizationStateInfos.put(fieldName, stateBytes);
+            }
+        } catch (Exception e) {
+            return readQuantizationStateInfos;
+        }
+        return readQuantizationStateInfos;
+    }
+
+    /**
+     * Reads an individual quantization state for a given field
+     * @param readConfig a config class that contains necessary information for reading the state
+     * @return quantization state
+     */
+    public static QuantizationState read(QuantizationStateReadConfig readConfig) throws IOException {
+        SegmentReadState segmentReadState = readConfig.getSegmentReadState();
+        String field = readConfig.getField();
+        String quantizationStateFileName = getQuantizationStateFileName(segmentReadState);
+        int fieldNumber = segmentReadState.fieldInfos.fieldInfo(field).getFieldNumber();
+
+        try (IndexInput input = segmentReadState.directory.openInput(quantizationStateFileName, IOContext.READ)) {
```

Comment:
This is a try without a catch, The same applies here, return null for a specific exception and throw for others

Reply:
Ideally we can have it , but this piece of code is getting called from KnnWeight only when QuantizationParm is not null.

Reply:
Same for this https://github.com/opensearch-project/k-NN/pull/1997#discussion_r1744563505

Reply:
File not found excpetion will never be for this case , as in KNN wight we already have check before calling this function